From cb6d7418c53e10bd4120891f6a2312d3ef839ca8 Mon Sep 17 00:00:00 2001
From: Conrad Kostecki <conikost@gentoo.org>
Date: Sun, 12 Dec 2021 23:04:58 +0100
Subject: [PATCH] CMakeLists.txt: update CCache handling (#16185)

This PR updated slightly the handling of CCache. With the current
implementation, when CCache is not being installed, CMake will complain,
that OPENRCT2_USE_CCACHE is not set.

This logic is slightly updated, so the option OPENRCT2_USE_CCACHE is always
being available and when this option is enabled (default ON), it will
search for CCache and warn, if it's not found.

The idea behind is, as in Gentoo, we never let CCache be used by the
package itself, instead, we enabled it globally. But this the old logic,
on systems, which don't have CCache installed, it will complain about
OPENRCT2_USE_CCACHE not being used, but we have to make sure, OpenRCT2
will never use by itself CCache and this message will be gone..

Signed-off-by: Conrad Kostecki <conikost@gentoo.org>
---
 CMakeLists.txt | 23 +++++++++++++----------
 1 file changed, 13 insertions(+), 10 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8d202cf5a9b8..566f3a8a9688 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,16 +8,19 @@ endif()
 # if it is available
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
-find_package(CCache)
-
-if (CCache_FOUND)
-    option(OPENRCT2_USE_CCACHE "Use CCache to improve recompilation speed (optional)" ON)
-    if (OPENRCT2_USE_CCACHE)
-        # Use e.g. "ccache clang++" instead of "clang++"
-        set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CCache_EXECUTABLE}")
-        set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK    "${CCache_EXECUTABLE}")
-    endif (OPENRCT2_USE_CCACHE)
-endif (CCache_FOUND)
+option(OPENRCT2_USE_CCACHE "Use CCache to improve recompilation speed (optional)" ON)
+
+if (OPENRCT2_USE_CCACHE)
+    find_package(CCache)
+
+    if (CCache_FOUND)
+         # Use e.g. "ccache clang++" instead of "clang++"
+         set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CCache_EXECUTABLE}")
+         set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK    "${CCache_EXECUTABLE}")
+    else()
+        message("Usage of CCache was enabled, but CCache was not found, so CCache is not being enabled.")
+    endif()
+endif (OPENRCT2_USE_CCACHE)
 
 if (APPLE)
     execute_process(COMMAND /usr/bin/uname -m OUTPUT_VARIABLE SYSTEM_MACOS_ARCH OUTPUT_STRIP_TRAILING_WHITESPACE)
